### DISCUSSION TITLE
Port with private dependencies?

### DISCUSSION THREAD

**Discussion by larne5** [microsoft/vcpkg#36495] [General]
Let's say I have a port with several customized dependencies from a private registry that are clashing with ports from vcpkg github (could be include paths, lib names). Is there a way to have these dependencies private to the port? Let's assume they are linked privately and statically.
Basically I would like to create a port that have dependencies not visible to the consumer of the port?

**Comment by dg0yt** [comment]
AFAICT such encapsulation doesn't generally exist in vcpkg. 

To some extent, it might be possible to assemble something in that direction with `vcpkg-port-config.cmake`. These files are only loaded for direct dependencies, and so they might inject customizations.
